rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java

```diff
@@ -79,41 +79,47 @@ public static Genesis loadGenesis(RskSystemProperties config, BigInteger initial
         }
     }
 
-    private static Map<ByteArrayWrapper, InitialAddressState> generatePreMine(RskSystemProperties config, BigInteger initialNonce, Map<String, AllocatedAccount> alloc){
-        Map<ByteArrayWrapper, InitialAddressState> premine = new HashMap<>();
+    private static Map<RskAddress, InitialAddressState> generatePreMine(RskSystemProperties config, BigInteger initialNonce, Map<String, AllocatedAccount> alloc){
+        Map<RskAddress, InitialAddressState> premine = new HashMap<>();
         ContractDetailsMapper detailsMapper = new ContractDetailsMapper(config);
+
         for (Map.Entry<String, AllocatedAccount> accountEntry : alloc.entrySet()) {
             if(!StringUtils.equals("00", accountEntry.getKey())) {
                 BigInteger balance = new BigInteger(accountEntry.getValue().getBalance());
                 BigInteger nonce;
+
                 if (accountEntry.getValue().getNonce() != null) {
                     nonce = new BigInteger(accountEntry.getValue().getNonce());
                 } else {
                     nonce = initialNonce;
                 }
+
                 AccountState acctState = new AccountState(nonce, balance);
                 ContractDetails contractDetails = null;
                 Contract contract = accountEntry.getValue().getContract();
+
                 if (contract != null) {
                     contractDetails = detailsMapper.mapFromContract(contract);
+
                     if (contractDetails.getCode() != null) {
                         acctState.setCodeHash(sha3(contractDetails.getCode()));
                     }
+
                     acctState.setStateRoot(contractDetails.getStorageHash());
                 }
-                premine.put(wrap(Hex.decode(accountEntry.getKey())), new InitialAddressState(acctState, contractDetails));
+
+                premine.put(new RskAddress(Hex.decode(accountEntry.getKey())), new InitialAddressState(acctState, contractDetails));
```

Comment:
Please use the other constructor: `new RskAddress(accountEntry.getKey())`

Reply:
done!